src/Compilers/CSharp/Portable/Binder/PatternVariableFinder.cs

```diff
@@ -165,6 +165,21 @@ public override void VisitArgument(ArgumentSyntax node)
                 return;
             }
 
+            switch (node.Identifier.
+                         Parent. // ArgumentSyntax
+                         Parent. // ArgumentListSyntax
+                         Parent. // invocation/constructor initializer
+                         Kind())
+            {
+                case SyntaxKind.InvocationExpression:
+                case SyntaxKind.ObjectCreationExpression:
+                case SyntaxKind.ThisConstructorInitializer:
+                case SyntaxKind.BaseConstructorInitializer:
+                    break;
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(node.Identifier.Parent.Parent.Parent.Kind());
```

Comment:
It probably makes sense to have more detailed info for this exception. 
We are intentionally crashing here if this code ends up being reachable (because of compile errors or some unforeseen reasons). More detailed message might make it easier to diagnose the crash than just unexpected Kind.

I mean - if we do work to detect the case, it makes sense to make message a bit more detailed.